README.md

```diff
@@ -20,6 +21,48 @@ mbed's test suite (codenamed Greentea) supports the *test supervisor* concept. T
 
 The basic host test only monitors the device's default serial port (the serial console or - in the future - console communication channel) for test result prints returned by the test runner in a specific and unique format. In other cases, a host test can, for example, judge from the test runner's console output if the test passed or failed. It all depends on the test itself: In some cases the host test can be a TCP server echoing packets from the test runner and judging packet loss. In other cases it can just check whether values returned from an accelerometer are actually valid (sane).
 
+# Writing host tests
+When writing a new host test for your module please bear in mind that:
+* You own the host test and you should write it the way so it can coexist with the same host tests ran by other processes such as Continuous Integration  systems, other host users etc.
+  * Note: If you work in isolation and your test environment if fully controlled by you (for example you queue all tasks calling host tests, or use global host unique socket port numbers) this rule doesn’t apply to you.
+* When writing host test using OS resources such as sockets, files, serial ports, peripheral devices like multi-meters /scopes etc. remember that those resources are indivisible!
+  * For example if you hardcode in your host test UDP port 32123 and use it for UDP server implementation  of your host test bear in mind that this port may be already used. It is your responsibility to react for this event and implement means to overcome it (if possible).
+So you would rather do somethins like this in your host test:
+```python
+import socket
+
+for port in range(32000, 32100):
+    try:
+        s.connect((HOSTNAME, PORT))
+        ...
+        # Got connection on 'port' let's conitnue
```

Comment:
typo - continue, sae line 57

typo - somethins  - line 30 

line 27 - environment if fully controlled -> environment is fully controlled

47 line - ``` # Oh my, none of 3200-32099 ports if available! ``` - > ``` # Oh my, none of 3200-32099 ports are available! ```